web/handlers.go

```diff
@@ -97,26 +87,51 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	token, tokenLocation := app.ParseAuthTokenFromRequest(r)
+
 	if len(token) != 0 {
 		session, err := c.App.GetSession(token)
+		csrfCheckPassed := false
+
+		// CSRF Check
+		if tokenLocation == app.TokenLocationCookie && h.RequireSession && !h.TrustRequester && r.Method != "GET" {
```

Comment:
I'm not too familiar with CSRF stuff, but why aren't we checking it for GET requests?

Reply:
CSRF attacks usually aim to execute an action as the victim user, so that requires a state-changing operation. State-changing operations shouldn't happen as form of GET requests. See [RFC2616 - Sec 9.1.1](https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.1.1) - For that reason OWASP also clarifies in their [CSRF cheat ](https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)_Prevention_Cheat_Sheet#Resources_that_need_to_be_protected_from_CSRF_vulnerability)sheet that only non-GET resources must be protected, since the only risk is state modification. Information disclosure is protected by the same origin policy.

In practice nothing stops us from expecting CSRF tokens as well for GET requests, but I don't see any need for that. Caching proxy might also cache the responses depending how aggressively they do, so it might also cause issues there.

Reply:
Makes sense